pkg/cloudprovider/providers/azure/azure_standard.go

```diff
@@ -236,11 +236,34 @@ func (az *Cloud) getSecurityRuleName(service *v1.Service, port v1.ServicePort, s
 		safePrefix := strings.Replace(sourceAddrPrefix, "/", "_", -1)
 		return fmt.Sprintf("shared-%s-%d-%s", port.Protocol, port.Port, safePrefix)
 	}
+	if sourceAddrPrefix == "" {
+		rulePrefix := az.getRulePrefix(service)
+		return fmt.Sprintf("%s-%s-%d", rulePrefix, port.Protocol, port.Port)
+	}
+	// ensure backward compatibility
 	safePrefix := strings.Replace(sourceAddrPrefix, "/", "_", -1)
 	rulePrefix := az.getRulePrefix(service)
 	return fmt.Sprintf("%s-%s-%d-%s", rulePrefix, port.Protocol, port.Port, safePrefix)
 }
 
+func (az *Cloud) getSourceAddressPrefix(sources []string) (string, []string, error) {
+	var sourceAddressPrefix string
+	var sourceAddressPrefixes []string
+	if len(sources) == 1 {
+		sourceAddressPrefix = sources[0]
+		sourceAddressPrefixes = []string{}
+	} else if len(sources) > 1 {
+		sourceAddressPrefix = ""
+		// this sort is used to ensure we get a predictable order to derive a name
+		// it is not an accurate sort
+		sort.Strings(sources)
+		sourceAddressPrefixes = sources
+	} else {
+		return sourceAddressPrefix, sourceAddressPrefixes, fmt.Errorf("sourceAddress should not be empty")
+	}
+	return sourceAddressPrefix, sourceAddressPrefixes, nil
```

Comment:
could we merge these two together into `sourceAddressPrefixes`? This could make it easier to understand.